adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/LoraProtocolAdapter.java

```diff
@@ -0,0 +1,636 @@
+        LOG.info("Starting Lora command consumers for tenants [{}] ...", commandEnabledTenants);
+        for (final String tenantId : commandEnabledTenants) {
+            scheduleStartLoraCommandConsumer(tenantId);
+        }
+    }
+
+    private void scheduleStartLoraCommandConsumer(final String tenantId) {
+        LOG.info("Starting Lora command consumer for tenant '{}' ...", tenantId);
+        startLoraCommandConsumer(tenantId).recover(x -> {
+            LOG.error("Error starting initial Lora command consumer for tenant [{}], retry in {} ms", tenantId,
+                    LORA_COMMAND_CONSUMER_RETRY_INTERVAL, x);
+            vertx.setTimer(LORA_COMMAND_CONSUMER_RETRY_INTERVAL, y -> scheduleStartLoraCommandConsumer(tenantId));
+            return Future.succeededFuture();
+        });
+    }
+
+    private Future<MessageConsumer> startLoraCommandConsumer(final String tenantId) {
+        return getCommandConnection().createCommandConsumer(tenantId, LORA_COMMAND_CONSUMER_DEVICE_ID,
+                receivedCommandContext -> commandConsumer(tenantId, receivedCommandContext), close -> {
+                    LOG.info("Lora command consumer closing initiated");
+                    closeCommandConsumer(tenantId, LORA_COMMAND_CONSUMER_DEVICE_ID);
+                }, LORA_COMMAND_CONSUMER_RETRY_INTERVAL);
+    }
+
+    @Override
+    protected void onStartupSuccess() {
+        scheduleStartLoraCommandConsumers();
+    }
+
+    private void commandConsumer(final String tenantId, final CommandContext receivedCommandContext) {
+        Tags.COMPONENT.set(receivedCommandContext.getCurrentSpan(), getTypeName());
+        final Command command = receivedCommandContext.getCommand();
+        final CommandData commandData = new CommandData();
+        commandData.setCommand(command);
+
+        getTenantConfiguration(tenantId, receivedCommandContext.getCurrentSpan().context()).compose(tenantObject -> {
+            if (tenantObject.isAdapterEnabled(getTypeName())) {
+                final String logMsg = "Adapter " + getTypeName() + " is enabled for tenant " + tenantId;
+                LOG.debug(logMsg);
+                return Future.succeededFuture();
+            } else {
+                final String logMsg = "Adapter " + getTypeName() + " is not enabled for tenant " + tenantId;
+                LOG.error(logMsg);
+                return Future.failedFuture(logMsg);
+            }
+        }).compose(adapterEnabledForTenant -> {
+            // we accept the message before sending it to the device in order to provide a consistent behavior
+            // across all protocol adapters, i.e. accepting the message only means that the message contains
+            // a valid command which we are willing to deliver see
+            // org.eclipse.hono.adapter.http.AbstractVertxBasedHttpProtocolAdapter.createCommandConsumer
+            receivedCommandContext.accept(1);
```

Comment:
IMHO we should at least try to send the command to the provider and accept or reject based on the provider's response ...